apps/site/components/Common/Search/index.tsx

```diff
@@ -1,61 +1,132 @@
+      getFormattedPath(path, pageSectionTitle),
+    path: ({ path, siteSection }: ResultMapPath) =>
+      siteSection.toLowerCase() === 'docs' ? `/${path}` : `/${locale}/${path}`,
+    section: 'siteSection',
+  };
 
   return (
     <>
-      <button
-        type="button"
-        onClick={openSearchBox}
-        className={styles.searchButton}
+      <OramaSearchButton
+        style={{ flexGrow: 1 }}
+        colorScheme={colorScheme}
+        themeConfig={themeConfig}
         aria-label={t('components.search.searchBox.placeholder')}
       >
-        <MagnifyingGlassIcon className={styles.magnifyingGlassIcon} />
-
         {t('components.search.searchBox.placeholder')}
-        <kbd
-          title={`${osCommandKey} K`}
-          className={classNames(styles.shortcutIndicator, {
-            'opacity-0': isOSLoading,
-          })}
-        >
-          <abbr>{osCommandKey} K</abbr>
-        </kbd>
-      </button>
-
-      {isOpen ? <WithSearchBox onClose={closeSearchBox} /> : null}
+      </OramaSearchButton>
+
+      <OramaSearchBox
+        index={{ api_key: ORAMA_CLOUD_API_KEY, endpoint: ORAMA_CLOUD_ENDPOINT }}
+        colorScheme={colorScheme}
+        themeConfig={themeConfig}
+        sourceBaseUrl={BASE_URL}
+        sourcesMap={sourceMap}
+        resultMap={resultMap}
+        facetProperty="siteSection"
+        linksTarget="_self"
+        highlightTitle={{
+          caseSensitive: false,
+          HTMLTag: 'b',
+          CSSClass: 'font-bold',
+        }}
+        searchParams={DEFAULT_ORAMA_QUERY_PARAMS}
+        suggestions={DEFAULT_ORAMA_SUGGESTIONS}
+        chatMarkdownLinkHref={({ href }) => {
+          if (!href) return href;
+
+          const currentPageHost = `${window.location.protocol}//${window.location.host}`;
+          const docsBaseUrl = `${currentPageHost}/docs`;
+          const lowerCaseHref = href.toLowerCase();
+
+          // Keep original link if URl should not have locale
+          if (
+            !lowerCaseHref.startsWith(currentPageHost) ||
+            lowerCaseHref.startsWith(docsBaseUrl)
+          ) {
+            return href;
+          }
+
+          const before = lowerCaseHref.slice(0, currentPageHost.length);
+          const after = lowerCaseHref.slice(currentPageHost.length);
+          return `${before}/${locale}${after}`;
+        }}
+        onAnswerSourceClick={event => {
+          event.preventDefault();
+
+          const currentPageHost = `${window.location.protocol}//${window.location.host}`;
+          const { path } = event.detail.source;
+
+          if (!path.startsWith('docs/')) {
+            window.open(`${currentPageHost}/${locale}/${path}`, '_blank');
+          } else {
+            window.open(`${currentPageHost}/${path}`, '_blank');
+          }
+        }}
+        onSearchResultClick={event => {
+          event.preventDefault();
+
+          const currentPageHost = `${window.location.protocol}//${window.location.host}`;
```

Comment:
Can we not use `window.` APIs here at all, I'm pretty sure @/navigation's useRouter and usePathname have everything you need. Using global environment-based APIs hinders the ability of us to optimise this in the future.

Reply:
Hi!
I'm sorry, but I could not find a proper way to get current host through neither usePathname nor useRouter.
Also checked the other methods from createNavigation. Would you mind to let me know what API should I use?

Another solution would be go back to use "BASE_URL" from constants.

Please let me know

Reply:
As mentioned before, you don't need to retrieve the host.